src/components/UserList.js

```diff
@@ -1,23 +1,66 @@
 import React from 'react';
 import UserItem from './UserItem';
+import { connect } from 'react-redux';
+import { addUser } from '../actions';
 
 class UserList extends React.Component {
+    constructor(props){
+        super(props)
+        this.state = {
+            input: ''
+        }
+    }
+    handleChange = event => {
+        this.setState({input: event.target.value})
+    }
+    handleSubmit = (event) => {
+        const {add, users} = this.props
+        event.preventDefault();
+        const newUser = event.target[0].value
+
+        if(newUser.length > 0){
+            if(users.length === 0){
+                add(newUser, 1)
+            }else{
+                const ids = users.map(element => element.id);
+                add(newUser, Math.max(...ids) + 1 )
+            }
+            this.setState({input: ''})
+        }
```

Comment:
Można by było też użyć `uuid()`, o którym było już wspominane - ale to też jest ok :)